client_app/src/components/shelter/ShelterDashboard.js

```diff
@@ -89,7 +89,7 @@ function ShelterDashboard() {
             <h4>Today's Roster</h4>
             <a href="/shift-details">View all</a>
           </div>
-          <ShiftContainer shiftDetails={shiftDetails} />
+          <ShiftContainer shiftDetails={shiftDetails.slice(0, 3)} /> 
```

Comment:
You shouldn't use slice here since the ShiftContainer will pass the shiftDetails to the Roster component. The Roster component will filter out the past shifts first before slicing. Please keep it like it is.